### PR TITLE
Add examples for setting up self-host via docker

### PR DESCRIPTION
**Pull request by MaKleSoft**

### CODE REVIEW

File: docs/examples/hosting/docker/basic/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+version: "3.7"
+services:
+    server:
+        image: padloc/server
+        build:
+            context: github.com/padloc/padloc.git#docs/docker-examples
```
Comment (MaKleSoft) [inline docs/examples/hosting/docker/basic/docker-compose.yml:6]: We'll have to remember to changes the ref to `v4` before merging, and to remove it before merging into master.

**Comment by MaKleSoft** [comment]
@BrunoBernardino Just wanted to give you a chance to look over these if you're interested, but don't feel obligated to do any testing etc. I ended up adding three examples of increasing complexity, simply because adding all three wasn't much more work than adding one and because I think working your way from simple to complex examples is the best way to actually understand whats going on.

File: docs/examples/hosting/docker/basic/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+0. Install [Docker](https://docs.docker.com/get-docker/) and
+   [Docker Compose](https://docs.docker.com/compose/install/):
+1. Clone or download this folder and `cd` into it.
+2. Start the server and pwa:
+
+    ```sh
+    docker-compose up
+    ```
+
+Thats it! The web app is now available at http://localhost:8080.
```
Comment (BrunoBernardino) [inline docs/examples/hosting/docker/basic/README.md:17]:
```suggestion
That's it! The web app is now available at http://localhost:8080.
```

**Review by BrunoBernardino — APPROVED** [review]
Docs look good, I read through the changes carefully but didn't pull down. Only suggested a typo I found.

**Pull request merged**